apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -6,15 +6,96 @@ import {
   CommandLineStringParameter,
   ICommandLineActionOptions
 } from '@rushstack/ts-command-line';
+import { SyncHook, AsyncParallelHook, AsyncSeriesHook } from 'tapable';
+import { performance } from 'perf_hooks';
 
 import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+import { CleanAction } from './CleanAction';
 
-export interface IBuildActionOptions extends IHeftActionBaseOptions {}
+export interface IBuildActionOptions extends IHeftActionBaseOptions {
+  cleanAction: CleanAction;
+}
+
+/**
+ * @public
+ */
+export class BuildPhaseHooksBase {
+  public readonly run: AsyncParallelHook = new AsyncParallelHook();
+}
+
+/**
+ * @public
+ */
+export interface IBuildPhase<TBuildPhaseHooks extends BuildPhaseHooksBase = BuildPhaseHooksBase> {
+  hooks: TBuildPhaseHooks;
+}
+
+/**
+ * @public
+ */
+export interface ISharedCopyStaticAssetsConfiguration {
+  /**
+   * File extensions that should be copied from the src folder to the destination folder(s)
+   */
+  fileExtensions?: string[];
+
+  /**
+   * Paths or globs that should be explicitly excluded. This takes precedence over paths listed in \"include\".
+   */
+  exclude?: string[];
+
+  /**
+   * Paths or globs that should be explicitly included.
```

Comment:
It's not possible to be a path "or" a glob.  For example, if the path is `rush-lib/assets/rush-init/[dot]travis.yml` then the `[` must be escaped, otherwise it will be misinterpreted as a glob character class.  Options that accept globs cannot easily receive normal strings without the caller having to import `glob-escape` and remember to escape.

Thus wherever we use globs, the name should probably call it out explicitly, e.g. `includeGlobs` instead of `include`.